cvat/apps/engine/views.py

```diff
@@ -548,41 +600,33 @@ def _get_rq_response(queue, job_id):
 
         return response
 
+    @staticmethod
     @swagger_auto_schema(method='get', operation_summary='Method provides a list of sizes (width, height) of media files which are related with the task',
         responses={'200': ImageMetaSerializer(many=True)})
     @action(detail=True, methods=['GET'], serializer_class=ImageMetaSerializer,
-        url_path='frames/meta')
-    def data_info(self, request, pk):
-        try:
-            db_task = self.get_object() # call check_object_permissions as well
-            meta_cache_file = open(db_task.get_image_meta_cache_path())
-        except OSError:
-            task.make_image_meta_cache(db_task)
-            meta_cache_file = open(db_task.get_image_meta_cache_path())
+        url_path='data/meta')
+    def data_info(request, pk):
+        data = {
+            'original_size': [],
+        }
+
+        db_task = models.Task.objects.prefetch_related('data__images').select_related('data__video').get(pk=pk)
+
+        if db_task.data.compressed_chunk_type == models.DataChoice.VIDEO:
+            media = [db_task.data.video]
+        else:
+            media = list(db_task.data.images.order_by('frame'))
+
+        for item in media:
+            data['original_size'].append({
+            'width': item.width,
+            'height': item.height,
+        })
 
-        data = literal_eval(meta_cache_file.read())
         serializer = ImageMetaSerializer(many=True, data=data['original_size'])
```

Comment:
Let's rename ImageMetaSerializer to DataMetaSerializer. Also Is it possible to change it? Let's have the following structure:
{
    size: ...,
    chunk_size: ...,
    start_frame: ...,
    stop_frame: ....
    ...
[{
    width: XXX,
    height: YYY,
    name: ZZZ
}, ...]
}